js/app/test/cancel_events.spec.ts

```diff
@@ -0,0 +1,46 @@
+import { test, expect } from "@gradio/tootils";
+
+test("when using an iterative function the UI should update over time as iteration results are received", async ({
+	page
+}) => {
+	const start_button = await page.locator("button", {
+		hasText: /Start Iterating/
+	});
+	const textbox = await page.getByLabel("Iterative Output");
+
+	let output_values: string[] = [];
+	let last_output_value = "";
+	let interval = setInterval(async () => {
```

Comment:
Previously relied on websocket events to check for value changes. Now just store all value changes until 8, and then make sure all values 1-7 are stored in array.